### PR TITLE
Add 2024-04-01 for StackHCI

### PR DESCRIPTION
**Pull request by NikitaSingh-ms**
## Description

Adding 2024-04-01 for StackHCI

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/StackHCI/StackHCI.Autorest/Az.StackHCI.psd1 [hunk 1]
```diff
@@ -11,7 +11,7 @@
...
+  FunctionsToExport = 'Add-AzStackHCIVMAttestation', 'Disable-AzStackHCIAttestation', 'Disable-AzStackHCIRemoteSupport', 'Enable-AzStackHCIAttestation', 'Enable-AzStackHCIRemoteSupport', 'Get-AzStackHciArcSetting', 'Get-AzStackHciCluster', 'Get-AzStackHciExtension', 'Get-AzStackHCILogsDirectory', 'Get-AzStackHCIRemoteSupportAccess', 'Get-AzStackHCIRemoteSupportSessionHistory', 'Get-AzStackHCIVMAttestation', 'Install-AzStackHCIRemoteSupport', 'Invoke-AzStackHciConsentAndInstallDefaultExtension', 'Invoke-AzStackHciExtendClusterSoftwareAssuranceBenefit', 'New-AzStackHciArcSetting', 'New-AzStackHciCluster', 'New-AzStackHciExtension', 'Register-AzStackHCI', 'Remove-AzStackHciArcSetting', 'Remove-AzStackHciCluster', 'Remove-AzStackHciExtension', 'Remove-AzStackHCIRemoteSupport', 'Remove-AzStackHCIVMAttestation', 'Set-AzStackHCI', 'Set-AzStackHciClusterRemoteSupport', 'Start-AzStackHciClusterLogCollection', 'Unregister-AzStackHCI', 'Update-AzStackHciCluster', '*'
```
Comment (priyjain358) [inline src/StackHCI/StackHCI.Autorest/Az.StackHCI.psd1:14]: Did you test out Set-AzStackHciClusterRemoteSupport', 'Start-AzStackHciClusterLogCollection' what's the output?

File: src/StackHCI/StackHCI.Autorest/README.md [hunk 2]
```diff
@@ -47,16 +47,16 @@ In this directory, run AutoRest:
...
 require:
   - $(this-folder)/../../readme.azure.noprofile.md
 # lock the commit
 input-file:
-  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-03-01/arcSettings.json
-  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-03-01/clusters.json
-  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-03-01/extensions.json
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/StackHCI/stable/2024-04-01/arcSettings.json
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/StackHCI/stable/2024-04-01/clusters.json
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/StackHCI/stable/2024-04-01/extensions.json
```
Comment (priyjain358) [inline src/StackHCI/StackHCI.Autorest/README.md:57]: Let's take the other resources as well: deploymentSettings, edgeDevices, securitySettings. 
Can you check with Amith Tallanki if we need to take updates related resources as well?

**Comment by priyjain358** [comment]
Update examples. 
And let's not add remote support and log collection commands yet.

File: src/StackHCI/StackHCI.Autorest/Az.StackHCI.psd1 (hunk 1 above)
Comment (NikitaSingh-ms) [reply src/StackHCI/StackHCI.Autorest/Az.StackHCI.psd1:14]: Yes i have added tests for both, you can check in the files

File: src/StackHCI/StackHCI.Autorest/README.md (hunk 2 above)
Comment (NikitaSingh-ms) [reply src/StackHCI/StackHCI.Autorest/README.md:57]: Sure

File: src/StackHCI/StackHCI.Autorest/docs/New-AzStackHciSecuritySetting.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+```
+
+## DESCRIPTION
+Create a security setting
+
+## EXAMPLES
+
+### Example 1:
+```powershell
+New-AzStackHciSecuritySetting -ClusterName 's-cluster' -ResourceGroupName 's-cluster-rg' -SName 'default'
```
Comment (priyjain358) [inline src/StackHCI/StackHCI.Autorest/docs/New-AzStackHciSecuritySetting.md:31]: Same comment for name

File: src/StackHCI/StackHCI.Autorest/docs/New-AzStackHciEdgeDevice.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+```
+
+## DESCRIPTION
+Create a EdgeDevice
+
+## EXAMPLES
+
+### Example 1:
+```powershell
+New-AzStackHciEdgeDevice -Name 'default' -ResourceUri "subscriptions/<subId>/resourceGroups/<test-rg>/providers/Microsoft.HybridCompute/machines/<test-node>"
```
[the remark this answers is not in the document]
Comment (NikitaSingh-ms) [reply src/StackHCI/StackHCI.Autorest/docs/New-AzStackHciEdgeDevice.md:27]: added directive for this

File: src/StackHCI/StackHCI.Autorest/docs/New-AzStackHciSecuritySetting.md (hunk 3 above)
Comment (NikitaSingh-ms) [reply src/StackHCI/StackHCI.Autorest/docs/New-AzStackHciSecuritySetting.md:31]: added directive for this

**Comment by VeryEarly** [comment]
this PR contains breaking change: -SoftwareAssurancePropertySoftwareAssuranceStatus removed.
Is it supposed to target next major release?

File: tools/AzPreview/AzPreview.psm1 [hunk 5]
```diff
@@ -0,0 +1,316 @@
+#
```
Comment (VeryEarly) [inline tools/AzPreview/AzPreview.psm1:1]: why this file needed

Comment (NikitaSingh-ms) [reply tools/AzPreview/AzPreview.psm1:1]: it got autogenerated while doing a local build, i will remove it

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**